_posts/2022-05-19-quinoa-modern-ui-with-no-hassles.adoc

```diff
@@ -0,0 +1,74 @@
+---
+layout: post
+title: 'Quinoa - A Quarkus extension to create Modern UI with no hassles'
+date: 2022-05-19
+tags: quinoa web ui react angular vue lit spa nodejs frontend extension
+synopsis: Quinoa extension takes care of all the Web UI build/wiring/dev-mode hassles to let you focus of your Quarkus web application logic.
+author: adamevin
+---
+:imagesdir: /assets/images/posts/quinoa
+
+image::quinoa.jpg[alt=Quinoa Bowl]
+
+Quinoa... To be honest I wasn't really a fan at first, but then my wife made them in a bowl like a salad with avocado and raw salmon. Now I really enjoy it with a touch of soy sauce. Let's not digress too much, I am not here to talk about healthy food in this post, this is about an extension to enjoy coding modern UI with Quarkus.
+
+== Get started
+
+Create a new Quinoa project (with a base Quinoa starter code):
+
+* With https://code.quarkus.io/?a=quinoa-bowl&j=17&e=io.quarkiverse.quinoa%3Aquarkus-quinoa[code.quarkus.io]
+* With the https://quarkus.io/guides/cli-tooling[Quarkus CLI]:
+[source,bash]
+----
+quarkus create app quinoa-app -x=io.quarkiverse.quinoa:quarkus-quinoa
+----
+
+Then start the live-coding *and navigate to http://0.0.0.0:8080/quinoa.html*:
+[source,bash]
+----
+quarkus dev
+----
+
+https://quarkiverse.github.io/quarkiverse-docs/quarkus-quinoa/dev/#getting-started[Here] is the Quinoa guide to go deeper and configure your app with any Node based Web UI framework such as React, Angular, Lit, Webpack, Rollup, ...
+
+=== A bit of context
+
+Alongside Quarkus, interacting with web frameworks such as React and Angular, is a natural need. Many users have been requesting a smooth integration since the early stages of Quarkus (https://github.com/quarkusio/quarkus/issues/588[#588]). This is why I created the Quinoa extension for Quarkus.
```

Comment:
```suggestion
Alongside Quarkus, interacting with web frameworks such as React and Angular is a natural need. Many users have been requesting a smooth integration since the early stages of Quarkus (https://github.com/quarkusio/quarkus/issues/588[#588]). This is why I created the Quinoa extension for Quarkus.
```